osxkeychain.go

```diff
@@ -46,41 +49,39 @@ type InternetPassword struct {
 	AuthType       AuthenticationType
 }
 
-// Error codes from https://developer.apple.com/library/mac/documentation/security/Reference/keychainservices/Reference/reference.html#//apple_ref/doc/uid/TP30000898-CH5g-CJBEABHG
-var (
-	ErrUnimplemented     = errors.New("Function or operation not implemented.")
-	ErrParam             = errors.New("One or more parameters passed to the function were not valid.")
-	ErrAllocate          = errors.New("Failed to allocate memory.")
-	ErrNotAvailable      = errors.New("No trust results are available.")
-	ErrReadOnly          = errors.New("Read only error.")
-	ErrAuthFailed        = errors.New("Authorization/Authentication failed.")
-	ErrNoSuchKeychain    = errors.New("The keychain does not exist.")
-	ErrInvalidKeychain   = errors.New("The keychain is not valid.")
-	ErrDuplicateKeychain = errors.New("A keychain with the same name already exists.")
-	ErrDuplicateCallback = errors.New("More than one callback of the same name exists.")
-	ErrInvalidCallback   = errors.New("The callback is not valid.")
-	ErrDuplicateItem     = errors.New("The item already exists.")
-	ErrItemNotFound      = errors.New("The item cannot be found.")
+type _OSStatus C.OSStatus
+
+// TODO: Fill this out.
+const (
+	errDuplicateItem _OSStatus = C.errSecDuplicateItem
```

Comment:
These changes to error handling look like a big improvement. Clearly you have a much better understanding of this than me :)

The only drawback I'm seeing is that there are no longer exported errors that users can match against. You can see this difference in the test file, where the test has to rely on internal methods to see what type of error was returned. I'd really like to keep that as I think it's good API design for a Go package like this.

Do you see any problems with doing something like this?

```go
var (
	ErrDuplicateItem = errorFromSecMessage(C.errSecDuplicateItem)
	// ...
)

var errCodeMap = map[C.OSStatus]error{
	C.errSecDuplicateItem: ErrDuplicateItem,
	// ...
}

func errFromSecMessage(errCode C.OSStatus) error {
	errorMessageCFString := C.SecCopyErrorMessageString(ke.errCode, nil)
	defer C.CFRelease(C.CFTypeRef(errorMessageCFString))

	errorMessageCString := C.CFStringGetCStringPtr(errorMessageCFString, C.kCFStringEncodingASCII)
	if errorMessageCString != nil {
		return C.GoString(errorMessageCString)
	}

	return fmt.Errorf("keychainError with unknown error code %d", ke.errCode)
}

func newKeychainError(errCode C.OSStatus) error {
	if errCode == C.noErr {
		return nil
	}
	if err, ok := errCodeMap[errCode]; ok {
		return err
	}
	return errFromSecMessage(errCode)
}
```

Not positive whether you can use something like `C.OSStatus` as a map key. But I think this would let us source error strings from actual keychain API calls, while still getting to present them with friendly Go exports.